lib/ansible/modules/mount_facts.py

```diff
@@ -0,0 +1,637 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+
+DOCUMENTATION = """
+---
+module: mount_facts
+version_added: 2.18
+short_description: Retrieve mount information.
+description:
+  - Retrieve information about mounts from preferred sources and filter the results based on the filesystem type and device.
+options:
+  devices:
+    description: A list of fnmatch patterns to filter mounts by the special device or remote file system.
+    default: ~
+    type: list
+    elements: str
+  fstypes:
+    description: A list of fnmatch patterns to filter mounts by the type of the file system.
+    default: ~
+    type: list
+    elements: str
+  sources:
+    description:
+      - A list of sources used to determine the mounts. Missing file sources (or empty files) are skipped.
+      - The C(mount_points) return value contains the first definition found for a mount point.
+      - Additional mounts to the same mount point are available from  C(aggregate_mounts) (if enabled).
+      - By default, mounts are retrieved from all of the standard locations, which have the predefined aliases V(all)/V(static)/V(dynamic).
+      - V(all) contains V(dynamic) and V(static).
+      - V(dynamic) contains V(/etc/mtab), V(/proc/mounts), V(/etc/mnttab), and the value of O(mount_binary) if it is not None.
+        This allows platforms like BSD or AIX, which don't have an equivalent to V(/proc/mounts), to collect the current mounts by default.
+        See the O(mount_binary) option to disable the fall back or configure a different executable.
+      - V(static) contains V(/etc/fstab), V(/etc/vfstab), and V(/etc/filesystems).
+        Note that V(/etc/filesystems) is specific to AIX. The Linux file by this name has a different format/purpose and is ignored.
+      - The value of O(mount_binary) can be configured as a source, which will cause it to always execute.
+        Depending on the other sources configured, this could be inefficient/redundant.
+        For example, if V(/proc/mounts) and V(mount) are listed as O(sources), Linux hosts will retrieve the same mounts twice.
+    default: ~
+    type: list
+    elements: str
+  mount_binary:
```

Comment:
IIUC the way this works, the mount binary wouldn't be able to accept args, since the value of `mount_binary` is also the "name" of its associated source. I'm wondering if it would make more sense to always use a fixed key like `mount` or `mount_binary` to refer to that in sources- then `mount_binary` could include args if needed?

Reply:
I could change sources to include a fixed key instead, but mount_binary is looked up using `get_bin_path` so args aren't supported for that reason. Should I just run the mount_binary as a command instead of looking up the binary? I was hesitant to support args, since it can impact the output format (like dragonflybsd has a -p option to format like fstab instead), which would not match the supported regexes for mount output.

Reply:
I updated sources to use the fixed key "mount" in the latest commit. I haven't added a way to pass args to the mount_binary.